File: sorting/insertion/insertion.go
package main

import "fmt"


func insertionSort(arr []int) {
	n := len(arr)
	for i := 1; i < n; i++ {
		key := arr[i]
		j := i - 1
		for j >=0 && arr[j] > key {
			arr[j+1] = arr[j]
			j = j - 1
		}
		arr[j+1] = key
	}
}

func main() {
	fmt.Println("Insertion Sort Algorithm")
	fmt.Print("Enter the number of elements: ")
	var n int
	fmt.Scanf("%d", &n)

	fmt.Println("Enter the elements")
	arr := make([]int, n)

	for i := 0; i < n; i++ {
		fmt.Scanf("%d", &arr[i])
	}

	insertionSort(arr)

	for i := 0; i < n; i++ {
		fmt.Printf("%d ", arr[i])
	}
}

File: sorting/insertion/insertion.py
def insertionSort(arr):
    """
    Sorts an array using the insertion sort algorithm.
    
    Parameters:
    arr (list): List of integers to be sorted.
    """
    n = len(arr)
    for i in range(1, n):
        key = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > key:
            arr[j + 1] = arr[j]
            j = j - 1

        arr[j + 1] = key


def main():
    print("Insertion Sort Algorithm")
    n = int(input("Enter the number of items: "))

    print("Enter the items separated by space")
    arr = [int(x) for x in input().split()]


    insertionSort(arr)

    print(arr)

if __name__ == "__main__":
    main()

